core/src/main/java/org/commonjava/maven/galley/io/checksum/ChecksummingOutputStream.java

```diff
@@ -131,17 +132,19 @@ public void write( final int data )
     }
 
     @Override
-    public void write (byte[] b, int off, int len)
-    throws IOException
+    public void write ( byte[] b, int off, int len )
+            throws IOException
     {
-        if ((off | len | (b.length - (len + off)) | (off + len)) < 0)
+        if ( ( off | len | ( b.length - ( len + off )) | ( off + len )) < 0 )
             throw new IndexOutOfBoundsException();
 
         super.write( b, off, len );
         size += len;
+        boolean isSubArray = 0 != off || len < b.length ;
+        byte[] sub = ( isSubArray ? Arrays.copyOfRange( b, off, len ) : b );
         for ( final AbstractChecksumGenerator checksum : checksums )
         {
-            checksum.update( b );
+            checksum.update( sub );
```

Comment:
I'd like to add a method update( final byte[] data, int offset, int len ) to AbstractChecksumGenerator, and call it here. This way we don't need to calc the sub-array by ourself. MessageDigest digester will offer it.

Reply:
Ok let's do that.